cl/custom_filters/templatetags/extras.py

```diff
@@ -120,3 +121,30 @@ def get(mapping, key):
 @register.simple_tag
 def random_int(a: int, b: int) -> int:
     return random.randint(a, b)
+
+
+# sourced from: https://stackoverflow.com/questions/2272370/sortable-table-columns-in-django
+@register.simple_tag
+def url_replace(request, value):
+    field = "order_by"
+    dict_ = request.GET.copy()
+    if field in dict_.keys():
+        if dict_[field].startswith("-") and dict_[field].lstrip("-") == value:
+            dict_[field] = value  # desc to asc
+        elif dict_[field] == value:
+            dict_[field] = "-" + value
+        else:  # order_by for different column
+            dict_[field] = value
+    else:  # No order_by
+        dict_[field] = value
+    return urlencode(sorted(dict_.items()))
+
+
+@register.simple_tag
+def sort_caret(request, value) -> SafeString:
+    current = request.GET.get("order_by", "*UP*")
```

Comment:
Sorry if I'm missing it, but how is `*UP*` used?

Reply:
It is not `None` which makes the following code simpler.  The text was chosen because nobody can name a field with an asterisk and the UP tells the developer that it is going to default to the up direction.

Reply:
Cool. I think I'd have used a comment, b/c this made me go hunting for how that value was used, but all good. Merging in a sec.

Reply:
Sorry.  It is a standard pattern from way back when.